server/src/main/java/org/opensearch/index/codec/composite/Composite99DocValuesReader.java

```diff
@@ -29,11 +60,123 @@
  */
 @ExperimentalApi
 public class Composite99DocValuesReader extends DocValuesProducer implements CompositeIndexReader {
-    private DocValuesProducer delegate;
+    private static final Logger logger = LogManager.getLogger(Composite99DocValuesReader.class);
+
+    private final DocValuesProducer delegate;
+    private IndexInput dataIn;
+    private ChecksumIndexInput metaIn;
+    private final Map<String, Tree> starTreeMap = new LinkedHashMap<>();
+    private final Map<String, CompositeIndexMetadata> compositeIndexMetadataMap = new LinkedHashMap<>();
+    private final Map<String, DocValuesProducer> compositeDocValuesProducerMap = new LinkedHashMap<>();
```

Comment:
One consumer for writer, hence there must be one producer for reader.
There must be only one file which gets initialized in the constructor after we form all fieldInfos of all the composite fields.